app/actions/workers/algorithms.test.js

```diff
@@ -0,0 +1,444 @@
+import expect, { spyOn, restoreSpies, createSpy } from 'expect'
+import sinon from 'sinon'
+import * as algorithms from './algorithms'
+// import Client from '../../utils/piazza-client'
+
+let _client, _handlers, _instance
+
+// describe("James' first bf test", function() {
+// 	it('should pass', function() {
+// 		expect(true).toBeTruthy();
+// 	});
+// });
+
+describe('Algorithms Worker', function() {
+
+	beforeEach(function() {
+		spyOn(console, 'debug');
+		spyOn(console, 'error');
+	})
+
+	afterEach(function() {
+		restoreSpies();
+	})
+
+	describe('start()', function() {
+
+		describe('program flow', function() {
+
+			beforeEach(function() {
+				_handlers = {
+					beforeFetch: createSpy().andReturn(false),
+					onFailure: createSpy().andReturn(false),
+					onTerminate: createSpy().andReturn(false),
+					onUpdate: createSpy().andReturn(false),
+					shouldRun: createSpy().andReturn(false)
+					};
+				_client = {
+					getServices: createSpy().andReturn(Promise.resolve([
+						{
+							resourceMetadata: {
+								description: '',
+								name: '',
+								metadata: {
+									Interface: ''
+								}
+							},
+							serviceId: '',
+							url: ''
+						}
+					]))
+				};
+			});
+
+			afterEach(function() {
+				algorithms.terminate()
+				// try {
```

Comment:
Try not to commit commented-out code.

I'm [guilty of this myself on occasion](https://github.com/venicegeo/bf-ui/blob/0b039089eb9acc959d7d364e1a85b4c95357f6c9/app/store/index.js#L261-L292), but the general idea is that if the code is important enough to retain, pull it into a snippet/gist and if it isn't, rip it out.  The 6-months-older you will appreciate your efforts. :)